### PR TITLE
add namespaces to plugin to make sure class names are unique for word…

### PR DESCRIPTION
**Pull request by richeduni**
…press directory inclusion

### CODE REVIEW

File: classes/class-blog-creator-factory.php [hunk 1]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+namespace EdLTI;
```
Comment (Sephster) [inline classes/class-blog-creator-factory.php]: Name spaces should reflect the directory.
```suggestion
namespace EdLTI\classes;
```

File: classes/class-blog-handler-factory.php [hunk 2]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+namespace EdLTI;
```
Comment (Sephster) [inline classes/class-blog-handler-factory.php]: Name spaces should follow the PSR 4 convention of reflecting the directory

```suggestion
namespace EdLTI\classes;
```

File: classes/class-blog-handler.php [hunk 3]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+namespace EdLTI;
```
Comment (Sephster) [inline classes/class-blog-handler.php]:
```suggestion
namespace EdLTI\classes;
```

File: classes/class-blog-handler.php [hunk 4]
```diff
@@ -35,11 +37,11 @@
 	/**
 	 * Get the WordPress role for a given LTI user role.
 	 *
-	 * @param User_LTI_Roles $roles
+	 * @param User_Lti_Roles $roles
```
Comment (Sephster) [inline classes/class-blog-handler.php]: LTI should probably be capitalised. Could we change this to be capitalised?

File: classes/class-config.php [hunk 5]
```diff
@@ -1,11 +1,14 @@
 <?php
+
+namespace EdLTI;
```
Comment (Sephster) [inline classes/class-config.php]:
```suggestion
namespace EdLTI\classes;
```

File: classes/class-config.php [hunk 6]
```diff
@@ -191,6 +194,9 @@ public function update_settings() {
 			update_site_option( 'lti_make_sites_private', 0 );
 		}
 
+		$is_helpline_url = sanitize_text_field( $_POST['is_helpline_url'] );
+		update_site_option( 'is_helpline_url', $is_helpline_url );
```
Comment (Sephster) [inline classes/class-config.php]: Is the `$is_helpline_url` used only once? If so, we could get rid of the variable

```suggestion
		update_site_option( 'is_helpline_url', sanitize_text_field( $_POST['is_helpline_url'] ) );
```

File: classes/class-course-blog-handler.php [hunk 7]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+namespace EdLTI;
```
Comment (Sephster) [inline classes/class-course-blog-handler.php]:
```suggestion
namespace EdLTI\classes;
```

File: classes/class-data.php [hunk 8]
```diff
@@ -1,11 +1,13 @@
 <?php
 
+namespace EdLTI;
```
Comment (Sephster) [inline classes/class-data.php]:
```suggestion
namespace EdLTI\classes;
```

File: classes/class-ed-lti.php [hunk 9]
```diff
@@ -1,4 +1,7 @@
 <?php
+
+namespace EdLTI;
```
Comment (Sephster) [inline classes/class-ed-lti.php]:
```suggestion
namespace EdLTI\classes;
```

File: classes/class-ed-lti.php [hunk 10]
```diff
@@ -58,11 +48,11 @@ public static function activate() {
 	 *
 	 * @return DataConnector
 	 */
-	private function lti_get_db_connector() {
+	private function get_db_connector() {
 		// phpcs:disable
 		return DataConnector::getDataConnector(
 			$this->wpdb->base_prefix,
-			new PDO( 'mysql:host=' . DB_HOST . ';dbname=' . DB_NAME, DB_USER, DB_PASSWORD )
+			new \PDO( 'mysql:host=' . DB_HOST . ';dbname=' . DB_NAME, DB_USER, DB_PASSWORD )
```
Comment (Sephster) [inline classes/class-ed-lti.php]: We should add a `use PDO` statement at the top of the file and remove the backslash here.

File: classes/class-ed-lti.php [hunk 11]
```diff
@@ -475,20 +464,20 @@ private function lti_redirect_user_to_blog_without_login( $blog_id ) {
 	 * @param string $alphabet
 	 *
 	 * @return string
-	 * @throws Exception
-	 * @throws InvalidArgumentException
+	 * @throws \Exception
```
Comment (Sephster) [inline classes/class-ed-lti.php]: Please change this to add a `use Exception` and remove the backslash.

File: classes/class-ed-lti.php [hunk 12]
```diff
@@ -475,20 +464,20 @@ private function lti_redirect_user_to_blog_without_login( $blog_id ) {
...
+	 * @throws \InvalidArgumentException
```
Comment (Sephster) [inline classes/class-ed-lti.php]: Please change this to add a `use InvalidArgumentException` and remove the backslash

File: classes/class-ed-lti.php [hunk 13]
```diff
@@ -475,20 +464,20 @@ private function lti_redirect_user_to_blog_without_login( $blog_id ) {
...
 	 */
 	private function random_string( $length, $alphabet ) {
 		if ( $length < 1 ) {
-			throw new InvalidArgumentException( 'Length must be a positive integer' );
+			throw new \InvalidArgumentException( 'Length must be a positive integer' );
```
Comment (Sephster) [inline classes/class-ed-lti.php]: Please change this to remove the backslash and add a `use InvalidArgumentException`

File: classes/class-ed-lti.php [hunk 14]
```diff
@@ -475,20 +464,20 @@ private function lti_redirect_user_to_blog_without_login( $blog_id ) {
...
 		}
 
 		$str = '';
 
 		$alphamax = strlen( $alphabet ) - 1;
 
 		if ( $alphamax < 1 ) {
-			throw new InvalidArgumentException( 'Invalid alphabet' );
+			throw new \InvalidArgumentException( 'Invalid alphabet' );
```
Comment (Sephster) [inline classes/class-ed-lti.php]: Please change this to remove the backslash and add a `use InvalidArgumentException`

File: classes/class-ed-tool-provider.php [hunk 15]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+namespace EdLTI;
```
Comment (Sephster) [inline classes/class-ed-tool-provider.php]:
```suggestion
namespace EdLTI\classes;
```

File: classes/class-ed-tool-provider.php [hunk 16]
```diff
@@ -22,11 +24,11 @@ public function onLaunch() {
 	 * Ensure errors in the LTI package are rendered correctly in WordPress
 	 *
 	 * @return void
-	 * @throws Exception
+	 * @throws \Exception
```
Comment (Sephster) [inline classes/class-ed-tool-provider.php]: Please add a `use Exception` at the top and remove this backslash

File: classes/class-ed-tool-provider.php [hunk 17]
```diff
@@ -22,11 +24,11 @@ public function onLaunch() {
...
 	 */
 	public function onError() {
 		wp_die( esc_html( $this->reason ) );
 
-		throw new Exception( $this->message );
+		throw new \Exception( $this->message );
```
Comment (Sephster) [inline classes/class-ed-tool-provider.php]: Please add a `use Exception` and remove this backslash

File: classes/class-ns-cloner-blog-creator.php [hunk 18]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+namespace EdLTI;
```
Comment (Sephster) [inline classes/class-ns-cloner-blog-creator.php]:
```suggestion
namespace EdLTI\classes;
```

File: classes/class-ns-cloner-blog-creator.php [hunk 19]
```diff
@@ -26,7 +28,7 @@ public function create( array $data ) {
 		$_POST['disable_addons'] = true;
 		$_POST['clone_nonce']    = wp_create_nonce( 'ns_cloner' );
 
-		$ns_site_cloner = new ns_cloner();
+		$ns_site_cloner = new \ns_cloner();
```
Comment (Sephster) [inline classes/class-ns-cloner-blog-creator.php]: Add a `use ns_cloner` statement and remove the backslash

File: classes/class-settings.php [hunk 20]
```diff
@@ -1,12 +1,14 @@
 <?php
 
+namespace EdLTI;
```
Comment (Sephster) [inline classes/class-settings.php]:
```suggestion
namespace EdLTI\classes;
```

File: classes/class-settings.php [hunk 21]
```diff
@@ -197,15 +199,15 @@ private function lti_do_validation() {
...
+	private function edit( $row = false ) {
 		$is_new = false;
 
 		if ( is_object( $row ) ) {
 			echo '<h3>Edit LTI</h3>';
 		} else {
 			echo '<h3>New LTI</h3>';
 
-			$row                  = new stdClass();
+			$row                  = new \stdClass();
```
Comment (Sephster) [inline classes/class-settings.php]: Add a use statement and remove the backslash

File: classes/class-student-blog-handler.php [hunk 22]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+namespace EdLTI;
```
Comment (Sephster) [inline classes/class-student-blog-handler.php]:
```suggestion
namespace EdLTI\classes;
```

File: classes/class-user-lti-roles.php [hunk 23]
```diff
@@ -1,11 +1,13 @@
 <?php
 
+namespace EdLTI;
```
Comment (Sephster) [inline classes/class-user-lti-roles.php]:
```suggestion
namespace EdLTI\classes;
```

File: classes/class-wp-blog-creator.php [hunk 24]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+namespace EdLTI;
```
Comment (Sephster) [inline classes/class-wp-blog-creator.php]:
```suggestion
namespace EdLTI\classes;
```

File: classes/interface-blog-creator-interface.php [hunk 25]
```diff
@@ -1,12 +1,14 @@
 <?php
 
+namespace EdLTI;
```
Comment (Sephster) [inline classes/interface-blog-creator-interface.php]:
```suggestion
namespace EdLTI\classes;
```

File: classes/class-ed-lti.php (hunk 10 above)
Comment (Sephster) [reply classes/class-ed-lti.php]: Hey Richard. I think this was missed in your initial pass at this. Could you take another look please? Many thanks

File: classes/class-config.php (hunk 6 above)
Comment (Sephster) [reply classes/class-config.php]: Hi Richard. This comment appears to have been missed. Could you take another look at it if you get a chance please? Many thanks!

File: CHANGELOG.md [hunk 26]
```diff
@@ -15,3 +15,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## 1.0.0 - 2018-10-30
 
 - First major release
+
+## 1.0.1 - 2018-11-16
+
+- Added namespaces to classes and added autoloading
```
Comment (Sephster) [inline CHANGELOG.md]: Hi Richard. Sorry, I should have provided more guidance with this change. As this change hasn't been committed into master yet, it should be listed under the _unreleased_ section. We also need to specify what type of change this is and I usually like referring back to the pull request for reference.

You can find more about the format that is expected by checking https://keepachangelog.com/en/1.0.0/

It should be something like

```
### Added
- Added namespaces to all classes (PR #19)

### Changed
- Removed lti prefixes from function names as no longer required (PR #19)
```

Comment (richeduni) [reply CHANGELOG.md]: fixed

File: classes/class-config.php (hunk 6 above)
Comment (Sephster) [reply classes/class-config.php]: @richeduni please could you address this comment? Thanks :+1:

File: classes/class-ed-lti.php (hunk 10 above)
Comment (Sephster) [reply classes/class-ed-lti.php]: @richeduni please could you address this comment? Thank you :)

File: ed-lti.php [hunk 27]
```diff
@@ -25,7 +25,12 @@
 along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
 
-require_once 'classes/class-ed-lti.php';
+namespace EdLTI;
+
+// Include the autoloader so we can dynamically include the rest of the classes.
```
Comment (Sephster) [inline ed-lti.php]: This comment should be above the section where we load the autoloader.

File: ed-lti.php [hunk 28]
```diff
@@ -25,7 +25,12 @@
...
+use EdLTI\classes\Ed_LTI;
```
Comment (Sephster) [inline ed-lti.php]: Does this use statement work? We haven't loaded the autoloader at this point so I thought it wouldn't do anything.

File: ed-lti.php (hunk 27 above)
Comment (richeduni) [reply ed-lti.php]: done

File: ed-lti.php (hunk 28 above)
Comment (richeduni) [reply ed-lti.php]: seems to be fine

File: classes/class-config.php (hunk 6 above)
Comment (richeduni) [reply classes/class-config.php]: fixed

**Pull request merged**